specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,316 @@
+Proper formatted message will be provided by MessageFormatter(Alerthandlers in present
+alerting module). The formatting will be specific to the tendrl object the alerting event is
+for and these messages will be added to payload of event.
+Note: The notification will display the message set by MessageFormatter.
+
+Once all the required details(grafana specific data and tendrl specific metadata) regarding the event
+is collected from all respective data sources(grafana and tendrl objects) it is passed as a
+message object to the node-agent socket. The alerting event is supposed to be notified to the user
+which will be done by the notifier component.
+
+Notifier functionalities are explained in https://github.com/Tendrl/specifications/pull/199
+
+=== Alternatives:
+
+None
+
+
+=== Data model impact:
+
+
+No changes in existing structure.
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The alerting evetns will now be stored in etcd under /alerting/alerts so the API will
+have to retreive the alerting events from the new path instead of fetching events from
+messages/events/ and then looking for alerting events i.e. events with priority as Notify 
+
+==== Notifications/Monitoring impact:
+
+None
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+Node-agent must be modified to insert the events received by Monitoring
+Integration in etcd under alerting/alerts/ instead of messages/events/.
+
+Monitoring integration adds scope ( global, cluster specific or node specific)
+of alert in the message metadata, this scope should be used by node-agent
+to place the event in the proper path in etcd.
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl/performance-monitoring impact:
+
+A  new component "Monitoring Integration" will be introduced to
+provide these functionalities.
+
+In this component:
+
+Create a new class called AlertHandler and run this class as separate gevent.
+AlertHandler will receive the alert event from socket and convert that alert
+as dictionary based on current alert structure and pass the alert as a message
+to the node-agent socket. (more details in implementation section)
+
+```
+To fetch details regarding particular alert from grafana:
+   http://{ip}:3000/api/alerts/{id}
+```
+
+==== Tendrl/alerting impact:
+
+Tendrl alerting system will be renamed as Notifier.
+Its main functionality would be to fetch the alert events and
+notify the respective user.
+
+=== Security impact:
+ 
+None
+ 
+=== Other end user impact:
+
+Users will be able to set alerts in Grafana itself and get notified for the same.
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+Need to create a end point to receive alerting event from grafana. Convert that alert event into current alerting structure. And pass the alert into node-agent socket.
+
+== Implementation:
+
+Create a new file called handler.py in performance monitoring.
```

Comment:
The implentation doesnt specify details of the "monitoring-integration" component

Also why are alerts being processed in performance monitoring component?